--- 
float64 grid_spacing
bool    success
